Source/ACE/Entity/Landblock.cs

```diff
@@ -509,6 +510,25 @@ public void EnqueueBroadcast(Position pos, float distance, params GameMessage[]
         }
 
         /// <summary>
+        /// Enqueues a message for global broadcast, thread safe
+        /// </summary>
+        /// <param name="msg"></param>
+        public void EnqueueGlobalBroadcast(params GameMessage[] msgs)
+        {
+            // Atomically checks and sets the broadcastQueued bit --
+            //    guarantees that if we need a broadcast it will be enqueued in the world-managers broadcast queue exactly once
+            if (Interlocked.CompareExchange(ref broadcastQueued, 1, 0) == 0)
+            {
+                WorldManager.BroadcastQueue.EnqueueAction(new ActionEventDelegate(() => SendGlobalBroadcasts()));
```

Comment:
Mogwai, I don't have time to verify for sure, but for the EnqueueBroadcast function you're probably right, that the order should be reversed.

However, I don't understand why this function was copy/pasted at all.  Why does this work need to be delayed?  For the location based broadcasts, its to ensure that two objects are not moving while updating.  For this, I don't there is a purpose to delaying the work.